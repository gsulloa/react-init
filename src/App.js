import React from 'react'
import { Switch, Route } from 'react-router-dom'
import Home from './screens/Home'

const App = () => (
  <Switch>
    <Route component={Home} exact path="/" />
    <Route render={() => <div>not found</div>} />
  </Switch>
)

export default App
