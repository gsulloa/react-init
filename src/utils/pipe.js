// https://medium.freecodecamp.org/pipe-and-compose-in-javascript-5b04004ac937
export default (...fns) => x => fns.reduce((v, f) => f(v), x)
