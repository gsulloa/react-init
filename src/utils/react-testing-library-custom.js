import React from 'react'
import { render as rtlRender } from 'react-testing-library'
import { MemoryRouter } from 'react-router-dom'
import withProvider from '../config/withProvider'
import pipe from './pipe'

const withMemoryRouter = initialEntries => withProvider(MemoryRouter, { initialEntries })

function render(ui, { initialEntries } = {}, ...rest) {
  const ComponentUi = () => ui
  const WithProviders = pipe(withMemoryRouter(initialEntries))(ComponentUi)
  return rtlRender(<WithProviders />, ...rest)
}

export * from 'react-testing-library'
export { render }
