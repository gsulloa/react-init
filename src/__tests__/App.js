import React from 'react'
import { render } from '../utils/react-testing-library-custom'
import App from '../App'

it('renders without crashing', () => {
  render(<App />)
})

it('renders welcome message', () => {
  const { getByText } = render(<App />)
  expect(getByText('Learn React')).toBeInTheDocument()
})

it('match snapshot', () => {
  const { container } = render(<App />)
  expect(container.firstChild).toMatchSnapshot()
})
