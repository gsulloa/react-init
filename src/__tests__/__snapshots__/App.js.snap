// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot at home 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-3 {
  text-align: center;
}

.emotion-2 {
  background-color: #282c34;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.emotion-0 {
  -webkit-animation: animation-0 infinite 20s linear;
  animation: animation-0 infinite 20s linear;
  height: 40vmin;
}

.emotion-1 {
  color: #61dafb;
}

<div
  class="emotion-3"
>
  <header
    class="emotion-2"
  >
    <img
      alt="logo"
      class="emotion-0"
      src="logo.svg"
    />
    <p>
      Edit
       
      <code>
        src/Home.js
      </code>
       
      and save to reload.
    </p>
    <a
      class="emotion-1"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <button
      type="button"
    >
      Emotion
       
      <code>
        css
      </code>
       
      example
    </button>
  </header>
</div>
`;

exports[`match snapshot at not found 1`] = `
<div>
  not found
</div>
`;
